src/lib/wordpress-provider/playground-cli/playground-server-process-child.ts

```diff
@@ -219,17 +183,30 @@ async function startServer(
 						code: `<?php require_once( '/wordpress/wp-load.php' ); update_option( 'WPLANG', '${ escapePhpString(
 							siteLanguage
 						) }' ); echo "Language set to: ${ escapePhpString( siteLanguage ) }"; ?>`,
-					},
-				],
-				...( args.blueprint.steps || [] ),
-			];
+					}
+				);
+			}
+
+			if ( serverOptions.siteTitle ) {
+				setupSteps.push( {
+					step: 'runPHP',
+					code: `<?php
+						require_once( '/wordpress/wp-load.php' );
+						update_option( 'blogname', '${ escapePhpString( serverOptions.siteTitle ) }' );
+						echo "Site title set to: ${ escapePhpString( serverOptions.siteTitle ) }";
+					?>`,
+				} );
+			}
 		}
 
+		args.blueprint.steps = [ ...setupSteps, ...( args.blueprint.steps || [] ) ];
+		args.blueprint.constants = { ...args.blueprint.constants, ...defaultConstants };
+
 		server = await runCLI( args );
 
-		if ( serverOptions.siteTitle || serverOptions.adminPassword ) {
+		if ( serverOptions.adminPassword ) {
 			try {
-				await setSiteOptions( server, serverOptions );
+				await setAdminPassword( server, serverOptions.adminPassword );
 			} catch {
 				console.warn(
 					'Failed to set site options, but the server started successfully. Please check your site error log in wp-content/debug.log for more details'
```

Comment:
Maybe we could also change the warning message here - since only the admin password change may fail here now.

For example:

```suggestion
					'Failed to set admin password, but the server started successfully. Please check your site error log in wp-content/debug.log for more details'
```